src/include/rados/librados.hpp

```diff
@@ -1318,7 +1318,7 @@ namespace librados
       const std::string& name,     ///< daemon name (e.g., 'gwfoo')
       const std::map<std::string,std::string>& metadata); ///< static metadata about daemon
     int service_daemon_update_status(
-      const std::map<std::string,std::string>& status);
+      std::map<std::string,std::string>&& status);
```

Comment:
@jdurgin @liewegas i understand this is part of the librados public API and is included by released tags. but my *guess* is that it is not used anywhere other than rgw, probably we can just change it w/o introducing a `service_daemon_update_status2()`?

Reply:
It's used by rbd-mirror and tcmu-runner (out-of-tree)

Reply:
... cancel that -- tcmu-runner only utilizes the C API version.

Reply:
oh, right! thanks, @dillaman:

- source code compatibility side: it's passing a temporary object. so it's passing rvalue reference already.
- binary side: we have "librados2 (= ${binary:Version})," in debian/control, and "librados2 = %{_epoch_prefix}%{version}-%{release}" in ceph.spec.

so we are safe here.